packages/tasit-action/src/TasitAction.test.js

```diff
@@ -241,18 +326,22 @@ describe("TasitAction.Contract", () => {
         confirmationFn();
       };
 
-      await subscription.on("confirmation", confirmationListener);
+      txSubscription.on("confirmation", confirmationListener);
 
-      const errorListener = error => {
-        errorFn();
-        subscription.removeAllListeners();
+      const errorListener = message => {
+        const { error, eventName } = message;
+        txSubscription.off("confirmation");
 
-        // FIXME: Assertion not working
-        //expect(1).to.equal(2);
-        expect(error.message).to.match(/uncle/);
+        // Note: This assertion will not fail the test case (UnhandledPromiseRejectionWarning)
```

Comment:
Yes, for now, the expected behavior when an eventListener throw an error is to emit an error event.
When an exception was thrown by an errorListener the user should handle with that otherwise a recursive loop will occur.
Is that right?

Reply:
That makes sense to me!

Reply:
Wait, but how does that relate to the assertion not failing the test case?

Reply:
(That is, how does your comment relate to the green line of code this comment is on?)

Reply:
I don't if I've done right, but I've split the comment:

```javascript
      const errorListener = message => {
        const { error, eventName } = message;

        // Note: This assertion will not fail the test case (UnhandledPromiseRejectionWarning)
        expect(error.message).to.equal(
          "Your message has been included in an uncle block."
        );

        // But asserting fake function, if that throws, test case will fail.
        errorFn();
      };
```

Reply:
Ah oops, I shouldn't have used the word comment twice to mean two different things.

In other words, I was asking if your message in this thread related to this comment in the code `// Note: This assertion will not fail the test case (UnhandledPromiseRejectionWarning)`

I think your message in this thread did not relate to that comment in the code. However, I get the sense that indeed, that comment in the code is still true. So I'm going to resolve this.

Thanks! Sorry I wasn't clearer in this thread.